android/src/com/mapswithme/maps/settings/SettingsPrefsFragment.java

```diff
@@ -820,31 +791,6 @@ private void removePreference(@NonNull String categoryKey, @NonNull Preference p
     category.removePreference(preference);
   }
 
-  @Override
-  public void onAttach(Context context)
-  {
-    super.onAttach(context);
-
-    if (!(context instanceof Activity))
-      return;
-
-    mPathManager.startExternalStorageWatching((Activity) context, new StoragePathManager.OnStorageListChangedListener()
```

Comment:
Что будет сейчас, когда юзер смотрит в настройки и достаёт/вставляет карточку?

Reply:
Its inconsistent and confusing atm (tested without this PR, F-Droid flavor, a9.0).

If maps are on SD card and looking at main settings or inside of storage settings:
1. eject SD
2. "Save maps to" or sd card storage option disappear
3. a moment later the app restarts itself
4. it loads to the map screen without any maps (even world), no maps in Downloads, "Save maps to" not visible
5. insert SD card - nothing happens
6. after manual app restart the app loads with all the maps, storage option is being reset back to SD card automatically

If a user was looking at the map screen, then:
1. eject SD
2. the app exits / crashes and doesn't restart
4. manual start loads to the download world map screen
5. exit app, insert SD card
6. after manual app restart the app loads with all the maps, storage option is being reset back to SD card automatically

Reply:
If maps are in internal memory card and looking at main settings:
1. eject SD
2. "Save maps to" option disappear
3. went back to the map then to the settings again, the app restarted
4. it loaded with all the maps and without "Save maps to" setting

After that my SD stopped being detected by the phone and I stopped testing :)

But its quite clear this behavior should be improved. I won't touch it in this PR I think.

Reply:
I've tested ejecting an SD while looking at main settings with my PR changes - the only difference is that "Save menu to" menu item doesn't disappear, but the app restarts and loads with no maps - just like without the PR.

Looking at the code the "listener" I've deleted justs re-scans available storages to be able to hide the "Save maps to" menu item, but doesn't change writable dir.

Watching removable media should be done on the app start and ejections should be handled gracefully globally by the app, not just inside of settings screen. I'll create a separate issue for that.

Reply:
> 3\. a moment later the app restarts itself

Is it a crash?

Reply:
Restarting/crashing is not a bad idea at all, because properly supporting graceful reload is a very complex task for the core. Let's focus on cases where the SD is already there (or is not there), and how to properly handle them.

1. For example, I moved maps to SD (btw, @vng do we check maps CRC after doing that? Let's create an issue to avoid surprises). Then for some reason, SD was ejected, then inserted back. Obviously, that OM should work as expected.
2. Maps are on SD. SD was ejected but not inserted back. The user starts the app and it asks to re-download everything. Right? What happens when I insert the SD with maps again? Probably, storage with newer or more maps should be chosen. But what to do with unused maps on unused storage? Should they be removed or left there "just in case"? What if user again tries to move maps?

Reply:
Not sure. I've added some logs to the relevant ticket.

Reply:
> Restarting/crashing is not a bad idea at all, because properly supporting graceful reload is a very complex task for the core. Let's focus on cases where the SD is already there (or is not there), and how to properly handle them.
> 
>     1. For example, I moved maps to SD (btw, @vng do we check maps CRC after doing that? Let's create an issue to avoid surprises). Then for some reason, SD was ejected, then inserted back. Obviously, that OM should work as expected.
> 
>     2. Maps are on SD. SD was ejected but not inserted back. The user starts the app and it asks to re-download everything. Right? What happens when I insert the SD with maps again? Probably, storage with newer or more maps should be chosen. But what to do with unused maps on unused storage? Should they be removed or left there "just in case"? What if user again tries to move maps?

Let's discuss it in #2450

Reply:
AFAIK, we don't check

Reply:
Please remove the check for `hasMoreThanOneStorage` and always display this option in the menu. All other changes are unrelated to the subject of PR and require additional testing.